src/simulation/dynamics/stateSpaceEffector/stateSpaceStateEffector.h

```diff
@@ -0,0 +1,93 @@
+/*
+ ISC License
+
+ Copyright (c) 2024, Autonomous Vehicle Systems Lab, University of Colorado at Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#ifndef STATE_SPACE_STATE_EFFECTOR_H
+#define STATE_SPACE_STATE_EFFECTOR_H
+
+#include <Eigen/Dense>
+#include "simulation/dynamics/_GeneralModuleFiles/stateEffector.h"
+#include "simulation/dynamics/_GeneralModuleFiles/stateData.h"
+#include "architecture/_GeneralModuleFiles/sys_model.h"
+#include "architecture/utilities/avsEigenMRP.h"
+#include "architecture/utilities/bskLogging.h"
+
+class StateSpaceStateEffector: public StateEffector, public SysModel {
+public:
+    StateSpaceStateEffector(int n);
+    ~StateSpaceStateEffector();
+    void Reset(uint64_t CurrentClock);
+    void UpdateState(uint64_t CurrentSimNanos);
```

Comment:
Does this need to be a SysModel? And if it does, do you need to populate UpdateState and Reset, or can you just leave them as the parent class implementation?

Reply:
It doesn't need to be a ``SysModel`` to run, but I use the ``logger`` function to grab the state variables for plotting, which is only accessible through the ``SysModel`` class. I don't see an alternative to plotting private variables, and I don't think making them public is a good idea either. However, I'm also not a fan of having a post-processing use case influence how a module is run... @patkenneally can you help a man out here?

Reply:
I am fine with it staying a SysModel

Reply:
@joaogvcarneiro what you have is sufficient for now. Changes coming will remove this multiple coupling of logging, sys_model, etc. We can update this after those changes.